src/components/header/renderer.tsx

```diff
@@ -157,14 +161,31 @@ class Header extends React.Component<Props, State> {
     this.setState({
       invalidUrl: !gistCommits.ok,
     });
+    if (filename.length === 0 && this.state.invalidUrl) {
+      this.setState({
+        invalidFilename: false,
+      });
+    }
+    if (revision.length === 0 && this.state.invalidUrl) {
+      this.setState({
+        invalidRevision: false,
+      });
+    }
     const responseGistCommits = await gistCommits.json();
     if (revision.length === 0) {
       revision = responseGistCommits[0].version;
+      this.setState({
+        invalidRevision: false,
+      });
+    } else {
+      const revGistCommits = await fetch(`https://api.github.com/gists/${gistId}/${revision}`);
+      this.setState({
+        invalidRevision: !revGistCommits.ok,
```

Comment:
how do you know that the id isn't wrong?

Reply:
The gist-url part (gistId) is checked before this. So all we need to do is check if revision is correct or not.